src/components/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { RadioButton as CoreRadioButton } from 'wix-ui-core/radio-button';
+import { RADIOBUTTON_DATA_KEYS } from './datahooks';
+import styles from './RadioButton.st.css';
+
+export interface RadioButtonChangeEvent
+  extends React.MouseEvent<HTMLDivElement> {
+  value: string;
+}
+
+export interface RadioButtonClickEvent
+  extends React.MouseEvent<HTMLDivElement> {
+  value: string;
+}
+
+export interface RadioButtonProps {
+  checked?: boolean;
+  disabled?: boolean;
+  value: string;
+  name?: string;
+  label: string;
+  onChange(event: RadioButtonChangeEvent | RadioButtonClickEvent): void;
+}
+
+interface DefaultProps {
+  checked: boolean;
+  disabled: boolean;
+  name: string;
+}
+
+/** RadioButton */
+export class RadioButton extends React.Component<RadioButtonProps> {
+  static displayName = 'RadioButton';
+  static defaultProps: DefaultProps = {
+    checked: false,
+    disabled: false,
+    name: '',
+  };
+
+  getDataAttributes() {
+    const { checked, disabled } = this.props;
+    return {
+      [RADIOBUTTON_DATA_KEYS.Checked]: checked,
+      [RADIOBUTTON_DATA_KEYS.Disabled]: disabled,
+    };
+  }
+
+  render() {
+    const {
+      checked,
+      disabled,
+      value,
+      label,
+      name,
+      onChange,
+      ...rest
+    } = this.props;
+    const radioBtnIcon = <span className={styles.checkmark} />;
```

Comment:
I would move this outside of the render method, so it won't be created on each render.
I think it can even be outside the element, at the top